client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,160 @@
+			return;
+		}
+
+		regeneratePostByEmail( selectedSiteId );
+	};
+
+	const email = address && address !== 'regenerate' ? address : '';
+	const labelClassName =
+		moduleUnavailable || regeneratingPostByEmail || ! active ? 'is-disabled' : undefined;
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						siteIsAtomic
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ ! siteIsAtomic }
+				/>
+				{ siteIsJetpack ? (
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="post-by-email"
+						label={ translate( 'Publish posts by sending an email' ) }
+						disabled={ isFormPending || moduleUnavailable }
+					/>
+				) : (
+					<ToggleControl
+						checked={ active }
+						disabled={ isFormPending }
+						label={ translate( 'Post by Email' ) }
+						onChange={ ( checked ) => {
+							if ( ! selectedSiteId ) {
+								return;
+							}
+
+							if ( checked ) {
+								activateModule( selectedSiteId, moduleSlug );
+							} else {
+								deactivateModule( selectedSiteId, moduleSlug );
+							}
+						} }
+					/>
+				) }
+
+				<div className="publishing-tools__module-settings site-settings__child-settings">
+					<FormLabel className={ labelClassName }>
+						{ translate( 'Send your new posts to this email address:' ) }
+					</FormLabel>
+					<ClipboardButtonInput
+						className="publishing-tools__email-address"
+						disabled={ regeneratingPostByEmail || ! active || moduleUnavailable }
+						value={ email }
+					/>
+					<Button
+						onClick={ handleRegenerate }
+						disabled={
+							isFormPending || regeneratingPostByEmail || ! active || !! moduleUnavailable
+						}
+					>
+						{ regeneratingPostByEmail
+							? translate( 'Regenerating…' )
+							: translate( 'Regenerate address' ) }
+					</Button>
+				</div>
+			</FormFieldset>
+		</>
+	);
+};
+
+export const PostByEmailSetting = connect(
+	( state: IAppState ) => {
+		const selectedSiteId = getSelectedSiteId( state ) || 0;
+		const siteIsJetpack = isJetpackSite( state, selectedSiteId );
+		const isAtomic = isSiteAutomatedTransfer( state, selectedSiteId );
+		const regeneratingPostByEmail = isRegeneratingJetpackPostByEmail( state, selectedSiteId );
+		const active = !! isJetpackModuleActive( state, selectedSiteId, moduleSlug );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, selectedSiteId );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			selectedSiteId,
+			moduleSlug
+		);
+
+		return {
+			selectedSiteId,
+			siteIsJetpack,
+			isAtomic,
+			regeneratingPostByEmail,
+			active,
+			moduleUnavailable: siteInDevMode && moduleUnavailableInDevMode,
+		};
+	},
+	{
+		activateModule,
+		deactivateModule,
+		regeneratePostByEmail,
+	}
```

Comment:
… and same as here https://github.com/Automattic/wp-calypso/pull/90489/files#r1604574276, we should use `useDispatch` from `calypso/state` instead of `connect`s `mapDispatchToProps`